### ISSUE TITLE
Imbalance with balance similar node groups

### ISSUE THREAD

**Issue by JulienBalestra** [kubernetes/autoscaler#2892]
In the current AWS implementation, for example, the comparaison of node templates vs real world nodes produce imbalance scale ups when using the option `--balance-similar-node-groups`.

For example, when having 3 ASGs (1 per zone) with one ASG sets to 0, the autoscaler is always going  to scale the ASG with real world nodes instead of balancing the under provisioned ASG (the one at 0).

I suppose this is due to how we build node template and compare it to the real world ones.

To avoid this issue, we've been running these _experimental_ changes:
* https://github.com/DataDog/autoscaler/blob/datadog-master/cluster-autoscaler/core/utils.go#L248-L356
* https://github.com/DataDog/autoscaler/commit/9da4ff41cbab0bd25012fc70e5a9ce9995c6c3ef

These _experimental_ changes have been running properly for few months and gave us the expected results.
I'd like to open a conversation, why we initially used a mix comparaison of **real and templated nodes**.

This is also a potential issue when users change the instance type of the running ASG (https://github.com/kubernetes/autoscaler/pull/2840).

Let me know if you need more details I might have forgotten.

**Comment by MaciekPytel** [comment]
At a glance, this basically always uses TemplateNodeInfo, right? This was discussed at length in https://github.com/kubernetes/autoscaler/pull/1021.

tl;dr TemplateNodeInfo is always wrong. Even in controlled and uniform env like GKE it's impossible to get it 100% right (and, believe me, I tried). If you ever have a pod that barely fits or barely doesn't fit on a node (within the margin of error of TemplateNodeInfo) and you use a patch that always uses TemplateNodeInfo the effects are devastating (including infinite scale-up and / or pods remaining pending forever). 

I think a possibly better solution to the original problem is to add logic, where if you have multiple ~identical NodeGroup you use an existing node in one of the groups as a template for all the groups (overriding only the zone label).
